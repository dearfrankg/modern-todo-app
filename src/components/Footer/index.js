import React from 'react';
import {
  arrayOf, any, func, string,
} from 'prop-types';
import { Link } from 'react-router-dom';

import { getItemsLeftLabel, getListStatus } from 'helpers';
import * as filters from 'redux/types';

const propTypes = {
  todos: arrayOf(any).isRequired,
  clearCompleted: func.isRequired,
  activeFilter: string.isRequired,
};

const Footer = ({ todos, clearCompleted, activeFilter }) => {
  const hasTodos = todos.length > 0;
  const hasCompletedItems = getListStatus(todos).completedItems > 0;

  const footerComponent = (
    <div>
      <span className="todo-count">
        {getItemsLeftLabel(todos)}
      </span>

      <ul className="filters">
        <li>
          <Link className={activeFilter !== filters.SHOW_ALL ? '' : 'selected'} to="#/">
            All
          </Link>
        </li>
        <li>
          <Link className={activeFilter !== filters.SHOW_ACTIVE ? '' : 'selected'} to="#/active">
            Active
          </Link>
        </li>
        <li>
          <Link
            className={activeFilter !== filters.SHOW_COMPLETED ? '' : 'selected'}
            to="#/completed"
          >
            Completed
          </Link>
        </li>
      </ul>

      {hasCompletedItems && (
        <button className="clear-completed" onClick={clearCompleted}>
          Clear completed
        </button>
      )}
    </div>
  );

  return (
    <footer className="footer">
      {hasTodos && footerComponent}
    </footer>
  );
};

Footer.propTypes = propTypes;

export default Footer;
