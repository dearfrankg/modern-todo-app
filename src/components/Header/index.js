import React from 'react';
import { string, func } from 'prop-types';

const propTypes = {
  value: string.isRequired,
  handleSubmit: func.isRequired,
  handleChange: func.isRequired,
};

const Header = ({ value, handleSubmit, handleChange }) => (
  <header className="header">
    <h1>
todos
    </h1>

    <input
      autoFocus
      placeholder="What needs to be done?"
      type="text"
      className="new-todo"
      value={value}
      onChange={handleChange}
      onKeyDown={handleSubmit}
    />
  </header>
);

Header.propTypes = propTypes;

export default Header;
