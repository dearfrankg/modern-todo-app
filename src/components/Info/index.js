import React from "react";

const Info = () => (
  <footer className="info">
    <p>Double-click to edit a todo</p>
    <p>
      Created using by <button href="#">Frank Gutierrez</button>
    </p>
    <p>
      Inspired by <button href="#">SÃ©rgio A. Kopplin</button>
    </p>
  </footer>
);

export default Info;
