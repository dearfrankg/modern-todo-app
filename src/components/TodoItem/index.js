import React from "react";
import { objectOf, any, func, bool, string } from "prop-types";
import classNames from "classnames";

const propTypes = {
  inputRef: func.isRequired,
  todo: objectOf(any).isRequired,
  toggleTodo: func.isRequired,
  editing: bool.isRequired,
  editText: string.isRequired,
  deleteTodo: func.isRequired,
  handleEdit: func.isRequired,
  handleKeyDown: func.isRequired,
  handleChange: func.isRequired,
  handleSubmit: func.isRequired
};

const TodoItem = ({
  inputRef,
  todo,
  toggleTodo,
  editing,
  editText,
  deleteTodo,
  handleEdit,
  handleKeyDown,
  handleChange,
  handleSubmit
}) => (
  <li
    key={todo.id}
    className={classNames({
      completed: todo.completed,
      editing
    })}
  >
    <div className="view">
      <input
        className="toggle"
        type="checkbox"
        checked={todo.completed}
        onChange={() => toggleTodo(todo)}
      />

      <label onDoubleClick={handleEdit}>{todo.text}</label>

      <button className="destroy" onClick={() => deleteTodo(todo.id)} />
    </div>

    <input
      ref={inputRef}
      className="edit"
      value={editText}
      onBlur={handleSubmit}
      onChange={handleChange}
      onKeyDown={handleKeyDown}
    />
  </li>
);

TodoItem.propTypes = propTypes;

export default TodoItem;
