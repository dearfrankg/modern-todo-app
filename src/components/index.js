import Footer from './Footer';
import Header from './Header';
import Info from './Info';
import TodoItem from './TodoItem';

export {
  Footer, Header, Info, TodoItem,
};
