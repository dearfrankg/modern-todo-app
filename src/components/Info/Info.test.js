import React from 'react';
import { shallow } from 'enzyme';

import { Info } from 'components';

describe('Info:', () => {
  it('should render correctly', () => {
    expect(shallow(<Info />)).toMatchSnapshot();
  });
});
