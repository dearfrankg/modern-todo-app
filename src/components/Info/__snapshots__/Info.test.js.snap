// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info: should render correctly 1`] = `
<footer
  className="info"
>
  <p>
    Double-click to edit a todo
  </p>
  <p>
    Created using by 
    <button
      href="#"
    >
      Frank Gutierrez
    </button>
  </p>
  <p>
    Inspired by 
    <button
      href="#"
    >
      SÃ©rgio A. Kopplin
    </button>
  </p>
</footer>
`;
